portfolio/src/main/webapp/my-travels/travels.js

```diff
@@ -1,29 +1,57 @@
-const placesCoords = {
-                    "berlin" : {lat: 52.521999, lng: 13.413016},
-                    "default" : {lat: 52.048947, lng: 20.442758},
-                    "disneyland" : {lat: 48.869486, lng: 2.780874},
-                    "krakow" : {lat: 50.061683, lng: 19.937111},
-                    "kolobrzeg" : {lat: 54.186629, lng: 15.570729},
-                    "matterhorn" : {lat: 45.938319, lng: 7.731824},
-                    "zurich" : {lat: 47.349534, lng: 8.491872}
-                   }
+function createDescription(content) {
+  return `<div id="content">${content}</div>`;
+}
 
 function setLocation(location) {
-  const coords = placesCoords[location];
-  const map = new google.maps.Map(
-      document.getElementById('map'), {zoom: 10, center: coords});
-  const marker = new google.maps.Marker({position: coords, map: map});
+  if (typeof setLocation.locationsInfo === 'undefined') {
+    setLocation.locationsInfo = getMap();
+  }
+
+  setLocation.locationsInfo.then(locationsInfo => {
```

Comment:
that looks... interesting. Can you explain that to me?

Reply:
Sure!! I'm quite excited with this solution : )
This is a feature in Javascript (which I discovered today). Since everything in JS is an object, it can have properties. 
In my function setLocation() I create locationsInfo property. If this is property is uninitialised (line 6 is check for that), it fetches JSON from server using getMap() function. Once it fetches the JSON, it binds this value to the property and each next setLocation() call won't fetch anything, saving server calls time.
It's better than global variables -- it works like static variables in C++.

Reply:
I edited the comment, adding word Promise. Sorry for the inconvenience, if you were reading it now.

Reply:
I see, thanks for clarifying :-)